java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java

```diff
@@ -602,6 +602,10 @@ private String putSuffixToEnd(String name, int ind, int suffixLength) {
 		nearestObjects.addAll(cityVillageManager.getClosestObjects(location.getLatitude(), location.getLongitude()));
 		//either we found a city boundary the street is in
 		for (City c : nearestObjects) {
+			if(!c.getType().storedAsSeparateAdminEntity())
+				//ignore "city" thar never store in city table, see of IndexCreator.java methods: writeCitiesIntoDb && parameters of calling createMissingCity
```

Comment:
Comment would be:
// ignore districts, boroughs, ... so we don't get streets missing in the final obf